IPython/core/interactiveshell.py

```diff
@@ -2781,6 +2784,15 @@ def run_ast_nodes(self, nodelist, cell_name, interactivity='last_expr',
         if not nodelist:
             return
 
+        if interactivity == 'last_expr_or_assign':
+            if isinstance(nodelist[-1], ast.Assign):
+                asg = nodelist[-1]
+                if len(asg.targets) == 1:
+                    nnode = ast.Expr(ast.Name(asg.targets[0].id, ast.Load()))
```

Comment:
Oh, sweet. Thanks. Does it also define a pretty formatter for ast objects that are returned ?